### ISSUE TITLE
String interpolation with "{ }" substitution 

### ISSUE THREAD

**Issue by benjaminapetersen** [cel-expr/cel-go#640]
**Change**
Since CEL seems to be commonly used for a single expression only and is frequently embedded in markup such as YAML documents, string interpolation would be a friendly syntactic sugar to add to the language via the strings extension.  Languages embedded within other languages often lack editor/IDE support, therefore some features that are only syntactic sugar can still be valuable. 

**Example**
```yaml
- Current:  'firstName + ":" + lastName'
- Revised:  "{firstName}:{lastName}"

- Current:  'firstName != "bob" ? firstName + ":" + lastName : firstName'
- Revised:  'firstName != "bob" ? "{firstName}:{lastName}" : firstName'
```
It's a minor change but reduces the number of operators in a series making the result more readable. 

**Alternatives considered**
Considered compared to the existing string manipulation features.

**Comment by jpbetz** [comment]
See https://github.com/google/cel-go/pull/617. The description is stale on that PR however. The implementation uses `%` style formatting.

cc @DangerOnTheRanger

**Comment by TristonianJones** [comment]
Closed as duplicate of #617, though please keep in mind issue #655
